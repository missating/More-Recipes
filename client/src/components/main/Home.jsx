import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// components
import { ConnectedSignup } from '../auth/Signup';
import { ConnectedSignin } from '../auth/Signin';
import { ConnectedTopRecipes } from '../recipe/TopRecipes';

export const Home = props => (
  <div>
    <div>
      <div className="bg flex">
        <div className="home-content center">
          <h1> More Recipes </h1>
          <p>A platform for you to share your awesome recipes </p>
          <div className="btn-group">
            {!props.authenticated && (
              <div>
                <a
                  href="#signup-modal"
                  data-toggle="modal"
                  data-target="#signup-modal"
                >
                  <button
                    type="button"
                    className="btn signup-button"
                  >
                    SIGN UP
                  </button>
                </a>
              </div>
            )}
            <ConnectedSignup />

            {!props.authenticated && (
              <div>
                <a
                  href="#signin-modal"
                  data-toggle="modal"
                  data-target="#signin-modal"
                >
                  <button
                    type="button"
                    className="btn signin-button"
                  >
                    SIGN IN
                  </button>
                </a>
              </div>
            )}
            <ConnectedSignin />
          </div>
        </div>
      </div>
      <div className="scroll-holder">
        <a
          href="#featured"
        >
          <i
            className="fas fa-angle-double-down fa-3x scroll-down"
          />
        </a>
      </div>
    </div>
    <ConnectedTopRecipes />
  </div>
);

Home.propTypes = {
  authenticated: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  authenticated: state.auth.isAuthenticated,
});

const ConnectedComponent =
  connect(mapStateToProps)(Home);
export { ConnectedComponent as ConnectedHome };

