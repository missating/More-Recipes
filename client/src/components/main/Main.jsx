import React from 'react';
import { Switch, Route } from 'react-router-dom';
import { ConnectedHome } from './Home';
import { ConnectedRecipes } from '../recipe/Recipes';
import { ConnectedEditRecipe } from '../recipe/EditRecipe';
import { ConnectedSingleRecipe } from '../recipe/SingleRecipe';
import { ConnectedProfile } from '../user/Profile';
import { ConnectedAddRecipe } from '../recipe/AddRecipe';
import { ConnectedUserRecipes } from '../user/UserRecipes';
import { ConnectedUserFavourites } from '../user/UserFavourites';
import NotFound from '../common/NotFound';


export const Main = () => (
  <div className="main-cont">
    <Switch>
      <Route exact path="/" component={ConnectedHome} />
      <Route path="/recipes" exact component={ConnectedRecipes} />
      <Route path="/recipes/view/:id" exact component={ConnectedSingleRecipe} />
      <Route path="/profile" exact component={ConnectedProfile} />
      <Route path="/recipe/add" exact component={ConnectedAddRecipe} />
      <Route path="/users/recipes" exact component={ConnectedUserRecipes} />
      <Route path="/recipe/edit/:id" exact component={ConnectedEditRecipe} />
      <Route
        path="/users/favourites"
        exact
        component={ConnectedUserFavourites}
      />
      <Route component={NotFound} />
    </Switch>
  </div>
);

export { Main as RouterMain };
