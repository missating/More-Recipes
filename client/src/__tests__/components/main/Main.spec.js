import React from 'react';
import toJSON from 'enzyme-to-json';
import { Main } from '../../../components/main/Main';

describe('Main component', () => {
  it('should render', () => {
    const wrapper = shallow(<Main />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

