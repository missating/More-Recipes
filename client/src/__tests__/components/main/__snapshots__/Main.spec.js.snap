// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component should render 1`] = `
<div
  className="main-cont"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/recipes"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/recipes/view/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/profile"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/recipe/add"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/users/recipes"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/recipe/edit/:id"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/users/favourites"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
